src/controllers/scripting/legacy/controllerscriptenginelegacy.cpp

```diff
@@ -943,7 +813,7 @@ std::shared_ptr<QQuickItem> ControllerScriptEngineLegacy::loadQMLFile(
         rootItem->setHeight(pScreen->quickWindow()->height());
     }
 
-    return rootItem;
+    return std::unique_ptr<mixxx::qml::QmlMixxxControllerScreen>(rootItem);
```

Comment:
I think in order to truly benefit from resource safety here, the unique_ptr wrapping needs to be done on the return value of `qmlComponent.createWithInitialProperties` (directly in the same statement) (potentially after a qobject_cast). I don't want to force you to do it, but I don't have the time right now. If you're up for it, feel free to do it now and if not, remind me on sunday and I will file a PR against your branch.

Reply:
yeah I'm just worried about forgetting about it. But you're right, lets postpone for now.